server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt

```diff
@@ -127,42 +127,66 @@ object Chapter {
+                }
+                if (existingChapter != null) {
+                    chaptersToUpdate.add(fetchedChapter.apply {
+                        sourceOrder = index + 1
+                        it.mangaId = mangaId
+                        realUrl = (source as? HttpSource)?.getChapterUrl(this)
+                    })
+                }
+            }
+        }
 
         transaction {
-            chapterList.reversed().forEachIndexed { index, fetchedChapter ->
-                val chapterEntry = ChapterTable.select { ChapterTable.url eq fetchedChapter.url }.firstOrNull()
-                if (chapterEntry == null) {
-                    ChapterTable.insert {
-                        it[url] = fetchedChapter.url
-                        it[name] = fetchedChapter.name
-                        it[date_upload] = fetchedChapter.date_upload
-                        it[chapter_number] = fetchedChapter.chapter_number
-                        it[scanlator] = fetchedChapter.scanlator
-
-                        it[sourceOrder] = index + 1
-                        it[fetchedAt] = now++
-                        it[ChapterTable.manga] = mangaId
-
-                        it[realUrl] =
-                            runCatching {
-                                (source as? HttpSource)?.getChapterUrl(fetchedChapter)
-                            }.getOrNull()
-                    }
-                } else {
-                    ChapterTable.update({ ChapterTable.url eq fetchedChapter.url }) {
-                        it[name] = fetchedChapter.name
-                        it[date_upload] = fetchedChapter.date_upload
-                        it[chapter_number] = fetchedChapter.chapter_number
-                        it[scanlator] = fetchedChapter.scanlator
-
-                        it[sourceOrder] = index + 1
-                        it[ChapterTable.manga] = mangaId
-
-                        it[realUrl] =
-                            runCatching {
-                                (source as? HttpSource)?.getChapterUrl(fetchedChapter)
-                            }.getOrNull()
+            if (chaptersToInsert.isNotEmpty()) {
+                ChapterTable.batchInsert(chaptersToInsert) { chapter ->
+                    this[ChapterTable.url] = chapter.url
+                    this[ChapterTable.name] = chapter.name
+                    this[ChapterTable.date_upload] = chapter.date_upload
+                    this[ChapterTable.chapter_number] = chapter.chapter_number
+                    this[ChapterTable.scanlator] = chapter.scanlator
+                    this[ChapterTable.sourceOrder] = chapter.sourceOrder
+                    this[ChapterTable.fetchedAt] = chapter.fetchedAt
+                    this[ChapterTable.manga] = chapter.mangaId
+                    this[ChapterTable.realUrl] = chapter.realUrl
+                }
+            }
+
+            if (chaptersToUpdate.isNotEmpty()) {
+                chaptersToUpdate.forEach { chapter ->
```

Comment:
according to this https://stackoverflow.com/questions/51015020/how-to-do-batch-updates it should be possible to do batch updates